src/odemis/driver/xt_client.py

```diff
@@ -2279,6 +2286,52 @@ def _moveTo(self, future: futures.Future, pos: Dict[str, float], rel: bool = Fal
                 # know the user that the move is not complete.
                 if future._must_stop.is_set():
                     raise CancelledError()
+
+                # The stage is not moving anymore, but we still want to wait until the position has been updated
+                # TODO: base the timeout on the stage movement time estimation (est. *10)
+                timeout = 10  #s
+                expected_end_time = time.time() + timeout  # s
+                check_pos = True
+                wait_duration = 20e-03  # s
+
+                # Check for presence of rx/t or rz/r keys in pos
+                # In FM mode, the stage will maintain its rotation and tilt, if it is requested to change
+                # it is a different imaging mode
+                # TODO: extend the behavior to other imaging modes
```

Comment:
I have trouble understanding how you can extend stage movement to imaging modes, is this meteor specific?

Reply:
I have added a better explanation in the code. 

To explain here - We are checking only x and y and not other axes because during overview acquisition x and y (also z which is indirectly checked due to y-z linkage) are repeatedly changed , keeping the rotation and tilt constant. For other imaging modes, like SEM there may be request to change tilt repeatedly or something similar. 

`                # TODO: extend the behavior if rotation, tilt axes also need to be checked.
                # So far checking the above is not required, as we have observed reporting stage position only in x/y
                # being compromised. The issue with case western system was that the stage metadata was not
                # updated in timely manner during the overview acquisition in FM mode (during which rotation, tilt are
                # not changed), resulting in misplaced tiles in the image display. Even when the stage has physically
                # moved to the target position but took time in updating its actual position through metadata
                # The issue is resolved by checking if only x and y stage metadata reports the target position properly`